src/backend/distributed/commands/common.c

```diff
@@ -272,3 +275,71 @@ PreprocessDropDistributedObjectStmt(Node *node, const char *queryString,
 
 	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
 }
+
+
+/*
+ * DropTextSearchDictObjectAddress returns object address for the drop TextSearchDict stmt.
+ */
+ObjectAddress
+DropTextSearchDictObjectAddress(Node *node, bool missing_ok)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+
+	ObjectAddress objectAddress = { InvalidOid, InvalidOid, 0 };
+
+	List *objName = NIL;
+
+	foreach_ptr(objName, stmt->objects)
+	{
+		char *schemaName = NULL;
+		char *tsdictName = NULL;
+		DeconstructQualifiedName(objName, &schemaName, &tsdictName);
+
+		if (!schemaName)
+		{
+			/*
+			 * Citus should not throw error for non-existing objects, let Postgres do that.
+			 * Otherwise, Citus might throw a different error than Postgres, which we don't want.
+			 */
+			Oid tsdictOid = get_ts_dict_oid(objName, missing_ok);
```

Comment:
what if the `missing_ok` passed here is `false`. Wouldn't we throw error?

lets try to make these functions more generic

Reply:
The same comment is used in other places as well. We should remove it (except a few cases) because we are using the generic missing_ok flag. We pass it as true when we call address methods.